File: examples/modulith-clinical-tool-jpa/src/test/java/io/zenwave360/example/clinicaltool/modules/termsandconditions/infrastructure/jpa/inmemory/AcceptedTermsAndConditionsRepositoryInMemory.java
package io.zenwave360.example.clinicaltool.modules.termsandconditions.infrastructure.jpa.inmemory;

import io.zenwave360.example.clinicaltool.modules.termsandconditions.core.domain.*;
import io.zenwave360.example.clinicaltool.modules.termsandconditions.core.outbound.jpa.AcceptedTermsAndConditionsRepository;
import java.math.*;
import java.time.*;
import java.util.*;

public class AcceptedTermsAndConditionsRepositoryInMemory extends InMemoryJpaRepository<AcceptedTermsAndConditions> implements AcceptedTermsAndConditionsRepository {
    @Override
    public java.util.Optional<AcceptedTermsAndConditions> findByUserId(Long userId) {
        return getEntities().values().stream().filter(e ->
             isSameValue(userId, readField(e, "userId")) 
        ).findFirst();
    }
}

File: examples/modulith-klinical-tool-jpa/src/test/kotlin/io/zenwave360/example/clinicaltool/modules/termsandconditions/inmemory/AcceptedTermsAndConditionsRepositoryInMemory.kt
package io.zenwave360.example.clinicaltool.modules.termsandconditions.inmemory

import io.zenwave360.example.clinicaltool.modules.termsandconditions.AcceptedTermsAndConditionsRepository
import io.zenwave360.example.clinicaltool.modules.termsandconditions.domain.*
import java.math.*
import java.time.*
import java.util.*

class AcceptedTermsAndConditionsRepositoryInMemory :
    InMemoryJpaRepository<AcceptedTermsAndConditions, Long>(), AcceptedTermsAndConditionsRepository {
    override fun findByUserId(userId: Long): AcceptedTermsAndConditions? {
        return getEntities().values.firstOrNull { e -> isSameValue(userId, readField(e, "userId")) }
    }
}
